### PR TITLE
Fix avatars

### PR DESCRIPTION
**Pull request by rachelweinberg12**

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -24,6 +24,12 @@
       "@/*": ["./*"]
     }
   },
-  "include": ["next-env.d.ts", "**/*.ts", "**/*.tsx", ".next/types/**/*.ts"],
```
Comment (akrolsmir) [inline tsconfig.json:27]: It looks like we didn't check in the script `add-avatar-urls.js`, so let's undo the changes here?

File: next.config.js [hunk 2]
```diff
@@ -4,11 +4,16 @@ const nextConfig = {
     appDir: true,
   },
   images: {
+    domains: [
```
Comment (akrolsmir) [inline next.config.js:7]: I think we ended up not needing this change, so we can remove `domains`?

File: components/user-link.tsx [hunk 3]
```diff
@@ -20,17 +21,14 @@ export function shortenName(name: string) {
 }
 
 export function UserAvatarAndBadge(props: {
-  name: string
```
Comment (akrolsmir) [inline components/user-link.tsx:23]: Thanks for doing this refactor!

File: app/edit-profile/edit-profile.tsx [hunk 4]
```diff
@@ -190,14 +205,17 @@ async function saveProfile(
...
-  id: string,
-  supabase: SupabaseClient
+  supabase: SupabaseClient,
+  userId: string,
+  avatarSlug: string,
+  avatar: File
 ) {
-  if (!avatar) return
-  const { data, error } = await supabase.storage
+  if (!avatar) {
```
Comment (akrolsmir) [inline app/edit-profile/edit-profile.tsx:213]: It looks like saveAvatar expects `File` in the function params, so this should never be null; we can get rid of this check.

File: app/edit-profile/edit-profile.tsx [hunk 5]
```diff
@@ -172,7 +180,13 @@ async function saveProfile(
   avatar: File | null,
   supabase: SupabaseClient
 ) {
-  saveAvatar(avatar, new_profile.id, supabase)
+  const avatarSlug = uuid()
```
Comment (akrolsmir) [inline app/edit-profile/edit-profile.tsx:183]: Could nest all this logic inside the if statement, so that you're not checking for nullness twice. (This also makes the code cleaner!)
```
if (avatar) {
  const avatarSlug = uuid()
  const avatarUrl = ...
  await saveAvatar(...)
}

File: app/edit-profile/edit-profile.tsx [hunk 6]
```diff
@@ -90,12 +93,17 @@ export function EditProfileForm(props: { profile: Profile }) {
       <label htmlFor="avatar">Choose a profile picture:</label>
       <div className="flex space-x-2">
         <div className="h-24 w-24">
-          <Avatar
-            username={username ? username : undefined}
-            id={profile.id}
-            noLink
-            size={24}
-          />
+          {!avatar && <Avatar profile={profile} noLink size={24} />}
```
Comment (akrolsmir) [inline app/edit-profile/edit-profile.tsx:96]: might be good to use ternary operator here:
```
avatar ? <Image> : <Avatar>
```

File: app/edit-profile/edit-profile.tsx [hunk 7]
```diff
@@ -10,6 +10,9 @@ import { useRouter } from 'next/navigation'
 import { InformationCircleIcon } from '@heroicons/react/20/solid'
 import { Profile } from '@/db/profile'
 import { Select } from '@/components/select'
+import uuid from 'react-uuid'
+import Image from 'next/image'
+import { isNullishCoalesce } from 'typescript'
```
Comment (akrolsmir) [inline app/edit-profile/edit-profile.tsx:15]: I don't think we're using this import?

File: app/edit-profile/edit-profile.tsx [hunk 8]
```diff
@@ -90,12 +93,17 @@ export function EditProfileForm(props: { profile: Profile }) {
...
-            noLink
-            size={24}
-          />
+          {!avatar && <Avatar profile={profile} noLink size={24} />}
+          {avatar && (
+            <Image
+              width={24}
+              height={24}
+              className="my-0 h-24 w-24 flex-shrink-0 rounded-full bg-white object-cover"
+              style={{ maxWidth: `6rem` }}
```
Comment (akrolsmir) [inline app/edit-profile/edit-profile.tsx:102]: Note, you can actually accomplish this directly in tailwind:
`max-w-[6rem]`

See https://tailwindcss.com/docs/adding-custom-styles#using-arbitrary-values

File: app/admin/page.tsx [hunk 9]
```diff
@@ -81,6 +82,8 @@ export default async function Admin() {
           ))}
         </tbody>
       </Table>
+      <h2>Fix Avatars</h2>
```
Comment (akrolsmir) [inline app/admin/page.tsx:85]: I'd leave this file unchanged instead of leaving the import/header/code commented out here.

**Review by akrolsmir — APPROVED** [review]

File: tsconfig.json (hunk 1 above)
Comment (rachelweinberg12) [reply tsconfig.json]: done

File: next.config.js (hunk 2 above)
Comment (rachelweinberg12) [reply next.config.js]: done

File: app/edit-profile/edit-profile.tsx (hunk 4 above)
Comment (rachelweinberg12) [reply app/edit-profile/edit-profile.tsx]: done

File: app/edit-profile/edit-profile.tsx (hunk 5 above)
Comment (rachelweinberg12) [reply app/edit-profile/edit-profile.tsx]: done

File: app/edit-profile/edit-profile.tsx (hunk 6 above)
Comment (rachelweinberg12) [reply app/edit-profile/edit-profile.tsx]: done

File: app/edit-profile/edit-profile.tsx (hunk 7 above)
Comment (rachelweinberg12) [reply app/edit-profile/edit-profile.tsx]: lol you are right, deleted

File: app/edit-profile/edit-profile.tsx (hunk 8 above)
Comment (rachelweinberg12) [reply app/edit-profile/edit-profile.tsx]: done

File: app/admin/page.tsx (hunk 9 above)
Comment (rachelweinberg12) [reply app/admin/page.tsx:85]: discussed in person: this code is actually broken and crashes page if commented in because of server vs. client component rules.

**Comment by akrolsmir** [comment]
Awesome, thanks!
